cmd/gateway/gcs/gateway-gcs.go

```diff
@@ -130,31 +135,71 @@ EXAMPLES:
 func gcsGatewayMain(ctx *cli.Context) {
 	projectID := ctx.Args().First()
 	if projectID == "" && os.Getenv("GOOGLE_APPLICATION_CREDENTIALS") == "" {
-		errorIf(errGCSProjectIDNotFound, "project-id should be provided as argument or GOOGLE_APPLICATION_CREDENTIALS should be set with path to credentials.json")
+		minio.ErrorIf(errGCSProjectIDNotFound, "project-id should be provided as argument or GOOGLE_APPLICATION_CREDENTIALS should be set with path to credentials.json")
 		cli.ShowCommandHelpAndExit(ctx, "gcs", 1)
 	}
 	if projectID != "" && !isValidGCSProjectIDFormat(projectID) {
-		errorIf(errGCSInvalidProjectID, "Unable to start GCS gateway with %s", ctx.Args().First())
+		minio.ErrorIf(errGCSInvalidProjectID, "Unable to start GCS gateway with %s", ctx.Args().First())
 		cli.ShowCommandHelpAndExit(ctx, "gcs", 1)
 	}
 
-	startGateway(ctx, &GCSGateway{projectID})
+	minio.StartGateway(ctx, &GCS{projectID})
```

Comment:
would be better if we could avoid calling StartGateway from here. i.e a gateway package should not call minio.StartGateway(), gateway package should just return a struct or an interface for a `New()` call and be done with it. But it will be a little tricky to implement it because `cli` package's conventions. we can relook at this later if there is a better way to organise the code.

Reply:
Yeah even i thought about it.  But it is very tricky let me think about it.